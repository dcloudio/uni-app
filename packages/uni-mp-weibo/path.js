const path = require('path')
module.exports = {
  src: path.join(__dirname, 'src'),
  lib: path.join(__dirname, 'lib')
}
