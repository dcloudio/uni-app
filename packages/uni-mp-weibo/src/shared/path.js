export function parsePath (path) {
  let hash = ''
  let query = ''

  const hashIndex = path.indexOf('#')
  if (hashIndex >= 0) {
    hash = path.slice(hashIndex)
    path = path.slice(0, hashIndex)
  }

  const queryIndex = path.indexOf('?')
  if (queryIndex >= 0) {
    query = path.slice(queryIndex + 1)
    path = path.slice(0, queryIndex)
  }

  return {
    path,
    query,
    hash
  }
}
