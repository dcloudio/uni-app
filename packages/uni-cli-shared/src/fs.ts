import fs from 'fs'
import path from 'path'

export function emptyDir(dir: string, skip: string[] = []) {
  for (const file of fs.readdirSync(dir)) {
    if (skip.includes(file)) {
      continue
    }
    // node >= 14.14.0
    fs.rmSync(path.resolve(dir, file), { recursive: true, force: true })
  }
}
