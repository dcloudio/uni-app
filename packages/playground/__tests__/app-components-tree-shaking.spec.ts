import fs from 'fs-extra'
import path from 'path'
import execa from 'execa'

const projectDir = path.resolve(__dirname, '../app-components-tree-shaking')

describe('app-components-tree-shaking playground', () => {
  jest.setTimeout(50 * 1000)
  const types = {
    'uni-app-x': ['build:app-android', 'build:app-ios'],
  }
  const distDir = path.resolve(projectDir, 'dist')
  if (fs.existsSync(distDir)) {
    fs.emptyDirSync(distDir)
  }
  Object.keys(types).forEach((type) => {
    const scripts = types[type]
    scripts.forEach((script) => {
      const mode = script.split(':')[0]
      const platform = script.split(':')[1]
      test(`${type} ${script}`, async () => {
        const outDir = path.resolve(distDir, mode, type, platform)
        console.log(`${type} npm run ${script} start`)
        await execa('npm', ['run', script], {
          cwd: projectDir,
          env: {
            ...process.env,
            UNI_OUTPUT_DIR: outDir,
            UNI_APP_X: type === 'uni-app-x' ? 'true' : 'false',
          },
        })
        console.log(`${type} npm run ${script} end`)
        const manifest = JSON.parse(
          fs.readFileSync(path.resolve(outDir, 'manifest.json'), 'utf-8')
        )
        // 里边的版本号会变更，移除影响
        delete manifest['uni-app-x']
        expect(JSON.stringify(manifest, null, 2)).toMatchSnapshot()
      })
    })
  })
})
