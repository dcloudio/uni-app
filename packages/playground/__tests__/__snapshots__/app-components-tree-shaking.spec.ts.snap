// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-components-tree-shaking playground uni-app-x build:app-android 1`] = `
"{
  "id": "__UNI__XXXXXXX",
  "name": "app-components-tree-shaking",
  "description": "",
  "version": {
    "name": "1.0.0",
    "code": "100"
  },
  "app": {
    "distribute": {
      "modules": {
        "uni-map-tencent": {},
        "uni-canvas": {}
      }
    }
  }
}"
`;

exports[`app-components-tree-shaking playground uni-app-x build:app-ios 1`] = `
"{
  "id": "__UNI__XXXXXXX",
  "name": "app-components-tree-shaking",
  "description": "",
  "version": {
    "name": "1.0.0",
    "code": "100"
  },
  "app": {
    "distribute": {
      "modules": {
        "uni-map-tencent": {},
        "uni-canvas": {}
      }
    }
  }
}"
`;
